tests/src/rules/no-extraneous-dependencies.js

```diff
@@ -154,5 +158,29 @@ ruleTester.run('no-extraneous-dependencies', rule, {
         message: '\'lodash.isarray\' should be listed in the project\'s dependencies, not optionalDependencies.',
       }],
     }),
+    test({
+      code: 'import "not-a-dependency"',
+      options: [{packagePath: path.join(__dirname, '../../../package.json')}],
+      errors: [{
+        ruleId: 'no-extraneous-dependencies',
+        message: '\'not-a-dependency\' should be listed in the project\'s dependencies. Run \'npm i -S not-a-dependency\' to add it',
+      }],
+    }),
+    test({
+      code: 'import "not-a-dependency"',
+      options: [{packagePath: path.join(__dirname, '../../../doesnt-exist.json')}],
```

Comment:
Brainstorming: The name of this property could be named also as `manifestFilePath`. https://en.wikipedia.org/wiki/Manifest_file